lib/ts/nextjs.ts

```diff
@@ -127,6 +128,215 @@ export default class NextJS {
             });
         };
     }
+
+    private static async commonSSRSession(
+        baseRequest: PreParsedRequest,
+        options?: VerifySessionOptions,
+        userContext?: Record<string, any>
+    ): Promise<{
+        session: SessionContainer | undefined;
+        hasToken: boolean;
+        hasInvalidClaims: boolean;
+        baseResponse: CollectingResponse;
+        nextResponse?: Response;
+    }> {
+        let baseResponse = new CollectingResponse();
+
+        try {
+            let session = await Session.getSession(baseRequest, baseResponse, options, userContext);
+            return {
+                session,
+                hasInvalidClaims: false,
+                hasToken: session !== undefined,
+                baseResponse,
+            };
+        } catch (err) {
+            if (Session.Error.isErrorFromSuperTokens(err)) {
+                return {
+                    hasToken: err.type !== Session.Error.UNAUTHORISED,
+                    hasInvalidClaims: err.type === Session.Error.INVALID_CLAIMS,
+                    session: undefined,
+                    baseResponse,
+                    nextResponse: new Response("Authentication required", {
+                        status: err.type === Session.Error.INVALID_CLAIMS ? 403 : 401,
+                    }),
+                };
+            } else {
+                throw err;
```

Comment:
Cause of this, in the docs, you would have to tell users to use withSession like:
```
try {
   return await withSession(...)
} catch (err) {
   // send 500 error
}
```

Reply:
And also surround getSSRSession with try / catch if not done already

Reply:
done